apps/examples/testcase/le_tc/kernel/tc_smart_sfs.c

```diff
@@ -0,0 +1,895 @@
+
+	memset(data, '\0', sizeof(data));
+	file_writetest_fprintf(input);
+	file_readtest_fgets(data);
+	TC_ASSERT_EQ("tc_fgets_fprintf_data121", strcmp(input, data), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_fgets_fprintf_data255
+ * @brief                :using fgets & fprintf on 255 bytes data.
+ * @API's covered        :fprintf, fgets
+ * @return               :void
+ */
+
+void tc_fgets_fprintf_data255(void)
+{
+	char *input = "abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh12345678abcdefgh1234567";
+
+	memset(data, '\0', sizeof(data));
+	file_writetest_fprintf(input);
+	file_readtest_fgets(data);
+	TC_ASSERT_EQ("tc_fgets_fprintf_data255", strcmp(input, data), 0);
+	TC_SUCCESS_RESULT();
+
+}
+
+/**
+ * @fn                   :tc_readtest_1byte
+ * @brief                :reading only first byte of the input data
+ * @API's covered        :fprintf, fgets
+ * @return               :void
+ */
+
+void tc_readtest_1byte(void)
+{
+	char *input = "abcdefgh12345678abcd";
+	size_t len;
+
+	memset(data, '\0', sizeof(data));
+	len = strlen(input);
+	file_writetest_fwrite(input, len);
+	file_readtest_byte(data, 1, 1);
+	memset(testdata, '\0', sizeof(testdata));
+	strncpy(testdata, input, 1);
+	TC_ASSERT_EQ("tc_readtest_1byte", strcmp(testdata, data), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_readtest_5byte
+ * @brief                :reading first few bytes of the input data
+ * @API's covered        :fwrite, fread
+ * @return               :void
+ */
+
+void tc_readtest_5byte(void)
+{
+	char *input = "abcdefgh12345678abcd";
+	size_t len;
+
+	memset(data, '\0', sizeof(data));
+	len = strlen(input);
+	file_writetest_fwrite(input, len);
+	file_readtest_byte(data, 1, 5);
+	memset(testdata, '\0', sizeof(testdata));
+	strncpy(testdata, input, 5);
+	TC_ASSERT_EQ("tc_readtest_5byte", strcmp(testdata, data), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_readtest_middlebyte
+ * @brief                :reading a single byte from the middle of input data
+ * @API's covered        :fwrite, fread
+ * @return               :void
+ */
+
+void tc_readtest_middlebyte(void)
+{
+	char *input = "abcdefgh12345678abcdefgh";
+	size_t len;
+
+	memset(data, '\0', sizeof(data));
+	len = strlen(input);
+	file_writetest_fwrite(input, len);
+	file_readtest_byte(data, 11, 1);
+	memset(testdata, '\0', sizeof(testdata));
+	strncpy(testdata, input + 10, 1);
+	TC_ASSERT_EQ("tc_readtest_middlebyte", strcmp(testdata, data), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                   :tc_readtest_middlebyte10
+ * @brief                :reading a few bytes from the middle of input data
+ * @API's covered        :fwrite, fread
+ * @return               :void
+ */
+
+void tc_readtest_middlebyte10(void)
```

Comment:
TC name should be follow our format.
format is [TC_TYPE]_[MODULE_NAME]_[FUNC_NAME]_[FUNC_TYPE].
[TC_TYPE] should be among [itc, utc and tc].
[MODULE_NAME] is fs_sfs.
[FUNC_NAME] is file_readtest_byte.
[FUNC_TYPE] shoule be among [p, n or none].

Reply:
Ok, I will modify them.